devito/types/sparse.py

```diff
@@ -94,6 +94,18 @@ def gridpoints(self):
         """
         raise NotImplementedError
 
+    def interpolate(self, *args, **kwargs):
+        """
+        Implement an interpolation operation from the grid onto the given sparse points
+        """
+        return Interpolation(self.interpolator, *args, **kwargs)
```

Comment:
would the API be neater as `self.interpolator.interpolate(*args, **kwargs)` ? this way, here you don't need to know about `Interpolation` and `Injection`

Reply:
It was like what you're suggesting until Mathias made me change it, in order to get better __repr__ for the objects. 

In any case, does this have to happen in this PR?

Reply:
Don;t get how it's related to __repr__.

Reply:
The only reason the classes had to become specific for interpolation and injection was to be able to have the repr specific to interpolation/injection.

Reply:
That's not incompatible, `self.interpolato.interpolate` returns and `Interpolation` object and `.inject` returns and `Injection` object, that's not against separating both.

Reply:
I believe I have fixed this. Resolve if you confirm.